sipa/blueprints/generic.py

```diff
@@ -162,7 +165,18 @@ def logout():
                 extra={'tags': {'user': current_user.uid}})
     logout_user()
     flash(gettext("Abmeldung erfolgreich!"), 'success')
-    return redirect(url_for('.index'))
+
+    resp = redirect(url_for('.index'))
+
+    cookie_name = current_app.config.get('REMEMBER_COOKIE_NAME', COOKIE_NAME)
+    if cookie_name in request.cookies:
+        resp.set_cookie(cookie_name, "", 0)
+
+    current_app.login_manager._clear_cookie(resp)
+    resp.set_cookie("session", "")
+    session.clear()
+
+    return resp
```

Comment:
I understand your frustration @sgeisler, I hope you see my comments as constructive criticism.

A better way or even the correct way™ to delete the session cookie would probably be to provide a custom [`SessionInterface`](http://flask.pocoo.org/docs/0.11/api/#flask.sessions.SessionInterface) to Flask that extends the [`SecureCookieSessionInterface`](http://flask.pocoo.org/docs/0.11/api/#flask.sessions.SecureCookieSessionInterface) and overrides its [`save_session`](http://flask.pocoo.org/docs/0.11/api/#flask.sessions.SessionInterface.save_session) method and deletes the cookie if the session is empty.

I haven't looked at the `SecureCookieSessionInterface` source code in detail to be sure that overriding `save_session` is the correct place, but it this class is definitive the right the place to look.